components/contribution-flow/StepDetailsCrypto.js

```diff
@@ -26,14 +27,55 @@ const messages = defineMessages({
   },
 });
 
-const StepDetailsCrypto = ({ onChange, data }) => {
+/*
+ * Calculates the approximate value for 1 unit of a given crypto currency in the collective currency.
+ * Uses the Cryptonator API (https://www.cryptonator.com/api/)
+ */
+const getCryptoExchangeRate = async (cryptoCurrency, collectiveCurrency) => {
+  if (!cryptoCurrency || !collectiveCurrency) {
+    return null;
+  }
+  try {
+    const response = await fetch(`https://api.cryptonator.com/api/ticker/${cryptoCurrency}-${collectiveCurrency}`);
+    const body = await response.json();
+    if (!body.success || body.error !== '') {
+      throw new Error(`Cryptonator Error: ${body.error}`);
+    }
+    return body.ticker.price;
+  } catch (error) {
+    // we don't want the user to see any errors; simply don't show the conversion amount
+    // eslint-disable-next-line no-console
+    console.error(error);
+    return null;
+  }
+};
+
+const StepDetailsCrypto = ({ onChange, data, currency }) => {
   const intl = useIntl();
-  const [currencyType, setCurrencyType] = useState(data.currency);
+  const [cryptoCurrencyType, setCryptoCurrencyType] = useState(data.currency);
```

Comment:
`cryptoCurrencyType` and `setCryptoCurrencyType` doesn't seem good names. What does `type` means?

`cryptoCurrency` `setCryptoCurrency` ?

Reply:
You later name it `cryptoCurrency` (`const storeCryptoExchangeRate = async (cryptoCurrency`)

Reply:
Or maybe `selectedCryptoCurrency` `setSelectedCryptoCurrency` ? More precise.